course/src/matrix3.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Matrix library
+ * Author: Agustin Alba Chicar, 2019
+ * Author: Gerardo Puga, 2020
+ * Author: Steven Desvars, 2020
+ * Copyright 2020 Ekumen"
+ */
+
+#include <isometry/matrix3.hpp>
+
+namespace ekumen {
+namespace math {
+
+Matrix3::Matrix3(const std::initializer_list<double>& list) {
+  if (list.size() != 9) {
+    throw std::out_of_range("Incorrect size of list");
+  }
+  auto it = list.begin();
+  for (int i = 0; i < 3; ++i) {
+    row_1_[i] = *it;
+    ++it;
+  }
+  for (int i = 0; i < 3; ++i) {
+    row_2_[i] = *it;
+    ++it;
+  }
+  for (int i = 0; i < 3; ++i) {
+    row_3_[i] = *it;
+    if (i != 2) {
+      ++it;
+    }
+  }
+}
+
+Vector3 Matrix3::operator[](const int i) const {
+  switch (i) {
+    case 0:
+      return row_1_;
+    case 1:
+      return row_2_;
+    case 2:
+      return row_3_;
+    default:
+      throw std::out_of_range("Operator out of range");
+  }
+}
+
+Vector3& Matrix3::operator[](const int i) {
+  switch (i) {
+    case 0:
+      return row_1_;
+    case 1:
+      return row_2_;
+    case 2:
+      return row_3_;
+    default:
+      throw std::out_of_range("Operator out of range");
+  }
+}
+
+Matrix3 Matrix3::kIdentity{Matrix3{1, 0, 0, 0, 1, 0, 0, 0, 1}};
+Matrix3 Matrix3::kOnes{Matrix3{1, 1, 1, 1, 1, 1, 1, 1, 1}};
+Matrix3 Matrix3::kZero{Matrix3()};
+
+bool Matrix3::operator==(const Matrix3& r_matrix) const {
+  return row_1_ == r_matrix.row_1_ && row_2_ == r_matrix.row_2_ &&
+         row_3_ == r_matrix.row_3_;
+}
+
+bool Matrix3::operator!=(const Matrix3& r_matrix) const {
+  return !(*this == r_matrix);
+}
+
+double Matrix3::det() const {
+  double first_value, second_value, third_value;
+  first_value = (*this)[0][0] *
+                ((*this)[1][1] * (*this)[2][2] - (*this)[1][2] * (*this)[2][1]);
+  second_value = (*this)[0][1] * ((*this)[1][0] * (*this)[2][2] -
+                                  (*this)[1][2] * (*this)[2][0]);
+  third_value = (*this)[0][2] *
+                ((*this)[1][0] * (*this)[2][1] - (*this)[1][1] * (*this)[2][0]);
+  return first_value - second_value + third_value;
+}
```

Comment:
You could directly use the `row_0_`, `row_1_` , `row_2_`  instead of `(*this)[0]` `(*this)[1]` `(*this)[2]`

Also don't declare here the variable before using it. You could do:
`const double first_vale = row_0_....`

Nit: Replace `first_value`, `second_value` and `third_value` by `A`, `B` and `C`.